EIPS/eip-00.md

```diff
@@ -0,0 +1,790 @@
+
+#### `eth_getTransactionReceipt`
+
+Accepts the hash of a virtual transaction that encapsulates the validation frames of the `AA_TX_TYPE` transaction.
+This transaction's ID is defined as (`AA_TX_TYPE transaction hash + 1`).
+
+If an AA transaction is included in a block, returns the following values in addition to the existing fields:
+
+| Name                       | Value                                                                        |
+|----------------------------|------------------------------------------------------------------------------|
+| sender                     | Address of the sender of this transaction                                    |
+| nonce                      | The transaction nonce value                                                  |
+| paymaster                  | Address of the Paymaster if it is paying for the transaction, null otherwise |
+| deployer                   | Address of the Deployer if it is included in the transaction, null otherwise    |
+| senderCreationGasUsed      | The amount of gas actually used by the sender deployment frame               |
+| senderValidationGasUsed    | The amount of gas actually used by the sender validation frame               |
+| paymasterValidationGasUsed | The amount of gas actually used by the paymaster validation frame            |
+
+Accepts hash of Transaction Type `AA_TX_TYPE`.
+
+If an AA transaction is included in a block, returns the following values in addition to the existing fields:
+
+| Name                            | Value                                                                                          |
+|---------------------------------|------------------------------------------------------------------------------------------------|
+| status                          | Either 1 (success) or 0 (failure) status of the execution frame                                |
+| executionGasUsed                | The amount of gas actually used by the execution frame                                         |
+| postPaymasterTransactionStatus  | Either 1 (success), 0 (failure), or `null` (did not run) status of the `postPaymasterTransaction` frame |
+| postPaymasterTransactionGasUsed | The amount of gas actually used by the paymaster `postPaymasterTransaction` frame                       |
+
+Note that the field `to` is not included as there is no clear `target` in an `AA_TX_TYPE` transaction.
+
+#### `eth_call`
+
+Accepts Transaction Type `AA_TX_TYPE` with all fields except `from` and `callData` optional.
+
+Returns the return value of [the `sender` execution frame](#sender-execution-frame).
+
+If provided with `paymasterData` and `deployerData` also executes the corresponding frame.
+
+If any of the frames reverts the call returns the revert data of each reverted frame.
+
+#### `eth_estimateGasAccountAbstraction`
+
+Accepts Transaction Type `AA_TX_TYPE` with fields `validationGasLimit`, `paymasterGasLimit`, `callGasLimit` optional.
```

Comment:
What would the node use during simulation if they are not provided?

For the gas limit values, it can set to max and then estimate them.  For fees we can't set it to max because the account (or paymaster) doesn't have sufficient funds to pay it, so the call would fail.  Besides, paymaster's validation (or account's validation if it's multi-tenant) might enforce a limit to prevent unreasonable fees from being used to drain its balance.  Setting it to an arbitrarily high value may trigger a validation failure in this case.

Reply:
@yoavw What do you think about temporarily "adding" the `gasLimit * maxFeePerGas` to the balance whenever the estimation is done? That's what we do at zkSync and it has worked well so far.

I think that most wallets won't limit the maximal `maxFeePerGas` and if they do, they should supply the limited `maxFeePerGas` themselves.

Reply:
@dancoombs I think it's fine to let the RPC node put something there if it's missing.  It might fail validation if the transaction uses a paymaster or an account that enforce some policy on fees, but in that case the wallet should be aware that it's using something of that sort and provide its own values. In the common case it'll work fine.

It just means that the RPC will have to make a distinction between how it checks transactions during estimation and during sending.  In the latter case it must verify that the fees are sufficiently high and that the account is sufficiently funded to pay them.

@StanislavBreadless that's a nice usability hack.  We'll have to add it either to the account's balance, or to the paymaster's if the paymaster is non-zero.

@forshtat @drortirosh WDYT about both things?  Making the fees optional in the RPC, and injecting a balance to the account or the paymaster during estimation?